src/rational/q/arithmetic/exp.rs

```diff
@@ -38,6 +68,57 @@ impl Q {
 
 #[cfg(test)]
 mod test_exp {
+    use super::*;
+
```

Comment:
write a test where you look at e^value of a relatively small value and of a large one.

Reply:
Found out that taylor polynomials has terrible accuracy (+- 10^306 for e^709) and changed a lot because of it. Therefore, I replaced the implementation with another approach.


- We have a test case for 1, which I would consider relatively small.
- Added a test for medium (300) exponent.
- Added a test for u32::MAX that validates that the result is within +-6%.
 (This was quite challenging since the free wolframalpha version does not like calculating with such large numbers. Additionally, ensuring a good performance (not 1-30 sec) of the test was also hard)


New algorithm: 
```
e^x = 2^(log_2(e^x)) = 2^(x*log_2(e))

// Divide x*log_2(e) into an integer part i and the remainder r
2^(i+r) = 2^i * 2^r
2^r ~= r+1 for r in [0,1]
```
There is a flint function to efficiently calculate `a * 2^x` since that's just bit shifting.

As a result: constant number of calculations :partying_face: :D